File: Data Structures & Algorithms/score-of-a-string/submission-1.py
class Solution:
    def scoreOfString(self, s: str) -> int:
        if len(s) == 1: return 0 
        p = 1
        score = 0
        while p < len(s):
            score += abs(ord(s[p]) - ord(s[p-1]))
            p += 1
        return score

File: Data Structures & Algorithms/last-stone-weight/submission-1.java
class Solution {
    public int lastStoneWeight(int[] stones) {
        PriorityQueue<Integer> maxHeap = new PriorityQueue<>((a,b) -> b - a);
        for(int n : stones) {
            maxHeap.offer(n);
        }
        
        while(maxHeap.size() > 1) {
            int x = maxHeap.poll();
            int y = maxHeap.poll();
            int weight = Math.abs(x-y);
            if(weight != 0) maxHeap.offer(weight);
        }

        if(maxHeap.size() == 1) {
            return maxHeap.poll();
        } 

        return 0;
    }
}
